carservice/src/main/java/com/senlainc/warsaw/tyurin/controller/OrderController.java

```diff
@@ -0,0 +1,108 @@
+package com.senlainc.warsaw.tyurin.controller;
+
+import com.senlainc.warsaw.tyurin.dto.OrderDto;
+import com.senlainc.warsaw.tyurin.entity.Order;
+import com.senlainc.warsaw.tyurin.exception.CriteriaNotFoundException;
+import com.senlainc.warsaw.tyurin.service.IOrderService;
+import com.senlainc.warsaw.tyurin.util.builder.OrderBuilder;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.List;
+
+@Controller
+@RequestMapping("orders")
+public class OrderController {
+
+    private final OrderBuilder orderBuilder;
+    private IOrderService orderService;
+
+    public OrderController(OrderBuilder orderBuilder,
+                           IOrderService orderService) {
+        this.orderService = orderService;
+        this.orderBuilder = orderBuilder;
+    }
+
+    @ResponseBody
+    @PostMapping()
+    public void create(@Valid @RequestBody OrderDto orderDto) {
+        Order order = orderBuilder.build(orderDto);
+        orderService.addOrder(order);
+    }
+
+    @ResponseBody
+    @GetMapping("{id}")
+    public OrderDto find(@PathVariable long id) {
+        return orderBuilder.build(orderService.getOrderById(id));
+    }
+
+    @ResponseBody
+    @DeleteMapping("{id}")
+    public void delete(@PathVariable long id) {
+        orderService.removeOrder(id);
+    }
+
+    @ResponseBody
+    @GetMapping("archived/sort/{sort}")
+    public List<OrderDto> getArchivedOrdersSortByCriteria(@PathVariable String criteria) throws CriteriaNotFoundException {
+        if (criteria.equals("price")) {
```

Comment:
Controllers should have minimal amount of business logic, it is better to move this on a service layer